src/Framework/PaymentGateways/PaymentGateway.php

```diff
@@ -497,4 +547,17 @@ private function isFunctionImplementedInGatewayClass(string $methodName): bool
 
         return ($reflector->getDeclaringClass()->getName() === get_class($this));
     }
+
+    /**
+     * Checks to see if the provided method is returning a specific command type as result.
+     * It always returns 'false' if the provided method doesn't explicitly declare the return type.
+     *
+     * @unreleased
+     */
+    private function isFunctionReturningCommand(string $methodName, PaymentCommand $command): bool
+    {
+        $reflector = new ReflectionMethod($this, $methodName);
+
+        return $reflector->getReturnType() ? $reflector->getReturnType()->getName() : '' === get_class($command);
```

Comment:
@glaubersilva looks like we can't use `$reflector->getReturnType()->getName()` until php 7.1 (we support 7.0 right now).  My other comment might end up removing this anyway.

<img width="905" alt="Screenshot 2023-02-23 at 4 26 06 PM" src="https://user-images.githubusercontent.com/10138447/221035118-facff26b-66e6-4921-b1df-d6ded1a76620.png">